pytest_xprocess.py

```diff
@@ -48,8 +48,9 @@ def pytest_runtest_makereport(item, call):
     if logfiles is None:
         return
     for name in sorted(logfiles):
-        content = logfiles[name].read()
-        if content:
-            longrepr = getattr(report, "longrepr", None)
-            if hasattr(longrepr, "addsection"):  # pragma: no cover
-                longrepr.addsection("%s log" % name, content)
+        with open(str(logfiles[name])) as f:
+            content = f.read()
```

Comment:
this is a breaking change, it changes from reading added data  to reading all data

Reply:
a pytest_unconfigure or sessionfinish hook should probably close the logfiles ,

Reply:
Thanks, had missed this one.